src/org/zaproxy/zap/extension/ascanrulesAlpha/XSLTInjection.java

```diff
@@ -0,0 +1,162 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+import org.zaproxy.zap.model.Vulnerabilities;
+import org.zaproxy.zap.model.Vulnerability;
+
+/**
+ * The XSLTInjection plugin identifies XSLT injection vulnerabilities with all the parameters in GET and POST
+ *
+ * @author CaptainFreak
+ */
+public class XSLTInjection extends AbstractAppParamPlugin {
+		
+	private static final Logger LOG = Logger.getLogger(XSLTInjection.class);
+
+	private static final String[] errorCausingPayloads={"<",">","\'","\""};
+	
+	private static final String[] vendorReturningPayloads={"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/>",
+															"system-property(\'xsl:vendor\')/>",
+															"\"/><xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--",
+															"<xsl:value-of select=\"system-property(\'xsl:vendor\')\"/><!--"
+															};
+	private static final String[] xsltVendors={"libxslt","Microsoft","Saxonica","Apache","Xalan","SAXON","Transformiix"};
+	
+	//TODO Add time based payloads (SSRF) and improve accuracy
+	
+	@Override
+	public int getId() {
+		return 90017;
+	}
+
+	@Override
+	public String getName() {		
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.name");
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {  
+		return technologies.includes(Tech.XML); 
+	}
+	
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString("ascanalpha.xsltinjection.xslt.refs");
+	}
+
+	@Override
+	public void init() {
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Initialising XSLT Scanner.");
+		}
+	}
+	
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+		try {
+			//Count positive Tests
+			boolean positiveBasedOnError=false; 
+			//error causing tests
+			for (String payload : errorCausingPayloads){
+				msg = getNewMsg();
+				setParameter(msg, param, payload);
+				sendAndReceive(msg);
+				if(!msg.getResponseBody().toString().contains(payload)){
+					positiveBasedOnError=true;
```

Comment:
Using your test files under xampp on windows.
This check doesn't seem to be robust enough. Based on the following request `http://localhost:81/xslt/xslt.php?name=%3C&email=me%40my.net` results in:
```html
<me@my.net
<br />
<b>Warning</b>:  DOMDocument::load(): StartTag: invalid element name in file:/C:/kingthorin/ZAP_DEV/xampp/htdocs/xslt/xslt.xsl, line: 3 in <b>C:\kingthorin\ZAP_DEV\xampp\htdocs\xslt\xslt.php</b> on line <b>31</b><br />
<br />
<b>Warning</b>:  XSLTProcessor::importStylesheet(): compilation error in <b>C:\kingthorin\ZAP_DEV\xampp\htdocs\xslt\xslt.php</b> on line <b>35</b><br />
<br />
<b>Warning</b>:  XSLTProcessor::importStylesheet(): xsltParseStylesheetProcess : empty stylesheet in <b>C:\kingthorin\ZAP_DEV\xampp\htdocs\xslt\xslt.php</b> on line <b>35</b><br />
<br />
<b>Warning</b>:  XSLTProcessor::transformToXml(): No stylesheet associated to this object in <b>C:\kingthorin\ZAP_DEV\xampp\htdocs\xslt\xslt.php</b> on line <b>37</b><br />
```
Which obviously contains the injection, but should set `positiveBasedOnError=true`. Yet at the same time obviously web content is going to contain &gt; &lt; and quotes.... I think you're going to have to come up with some error string to check the response for (and will probably likely need to pre-check the original response for those as well, like this: https://github.com/zaproxy/zap-extensions/pull/1342)

Reply:
Note: I did quickly code past this to test further and the second check does return `libxslt`.

Reply:
@CaptainFreak this is the main hurdle, if you want to tackle this part (or share your ideas). I could probably deal with tweaking the other out standing things.